### PR TITLE
fix(webdav): handle mixed property statuses in multi-status responses

### PR DESCRIPTION
**Pull request by nako-ruru**
#### What is the purpose of this change?

This PR fixes a bug in the WebDAV backend where directories or files could randomly "disappear" from listings due to strict and fragile multi-status code parsing.

##### The Problem
A regression was recently introduced when a new branch for explicit property fetching (`useStandardProps`) was added to the `PROPFIND` request body:

```go
// The recently added branch that triggered the issue:
if f.hasOCMD5 || f.hasOCSHA1 {
   opts.Body = bytes.NewBuffer(owncloudProps)
} else if f.useStandardProps {
   opts.Body = bytes.NewBuffer(standardProps) // 👈 This sends explicit XML props
}
```
When rclone queries multiple specific properties via `standardProps `(e.g., `displayname`, `getcontentlength`, `resourcetype`), `RFC 4918` fully allows standard-compliant WebDAV servers to return mixed statuses within a single item. For example, a directory does not have a content length, so the server legally returns `404 Not Found` for that specific property, alongside a `200 OK` for the resource type itself.

Since `RFC 4918` does not mandate the order of `<propstat>` XML elements, some standard-compliant servers return the `404 Not Found` block before the `200 OK` block.

Currently, `StatusOK()` relies on `p.Code()`, which blindly checks only the first element (`p.Status[0]`):
```Go

// Old fragile logic path inside p.Code()
match := parseStatus.FindStringSubmatch(p.Status[0])
```
If `404` happens to be the first status in the slice, `p.Code()` returns `0`, causing `StatusOK()` to return `false`. As a result, `listAll()` executes a continue and silently skips valid directories, causing them to randomly "disappear" from listings since the use_standard_props feature was introduced.
##### The Fix

To safely fix this without breaking other methods (like `readMetaDataForPath`) that rely on the original behavior of `p.Code()`, we decouple `StatusOK()` and update it to iterate through the entire `p.Status` slice:
Keep `p.Code()` intact to avoid breaking historical metadata checks (such as checking for `425` Too Early in 0-RTT scenarios).

Upgrade `StatusOK()` with a dedicated `2xx` regex pattern to scan all statuses. As long as any single status indicates success (`2xx`), the item is considered valid.

```Go

var parseStatus2XX = regexp.MustCompile(`HTTP/[\d.]+\s+2\d{2}`)

// StatusOK examines the Status and returns an OK flag
func (p *Prop) StatusOK() bool {
	// Assume OK if no statuses received
	if len(p.Status) == 0 {
		return true
	}
	for _, statusStr := range p.Status {
		if parseStatus2XX.MatchString(statusStr) {
			return true
		}
	}
	return false
}
```
#### Was the change discussed in an issue or in the forum before?

No, this was discovered during local integration testing with a standard-compliant WebDAV server that orders `404` property statuses before `200` statuses in the XML payload.

#### Checklist

- [x] I have read the [contribution guidelines](https://github.com/rclone/rclone/blob/master/CONTRIBUTING.md#submitting-a-new-feature-or-bug-fix).
- [x] I have added tests for all changes in this PR if appropriate.
- [x] I have added documentation for the changes if appropriate.
- [x] All commit messages are in [house style](https://github.com/rclone/rclone/blob/master/CONTRIBUTING.md#commit-messages).
- [x] I'm done, this Pull Request is ready for review :-)

### CODE REVIEW

File: backend/webdav/api/types.go [hunk 1]
```diff
@@ -98,21 +98,19 @@ func (p *Prop) Code() int {
 	return code
 }
 
+// Parse a status of the form "HTTP/1.1 200 OK" or "HTTP/1.1 200"
+var parseStatus2XX = regexp.MustCompile(`HTTP/[\d.]+\s+2\d{2}`)
```
Comment (ncw) [inline backend/webdav/api/types.go]: This regex is unanchored so could theoretically match in the middle of a string. Unlikely though but worth fixing.

File: backend/webdav/api/types.go [hunk 2]
```diff
@@ -98,21 +98,19 @@ func (p *Prop) Code() int {
...
+
 // StatusOK examines the Status and returns an OK flag
```
Comment (ncw) [inline backend/webdav/api/types.go]: ...if there is at least one 2xx response

**Review by ncw — CHANGES_REQUESTED** [review]
I think this looks like a good fix - thanks.

I put a couple of minor nits inline.

A test would be nice - we don't seem to have any at the moment!

**Comment by nako-ruru** [comment]
Hi @ncw,

Updated! I've anchored the regex, refactored the comments to explicitly focus on the 2XX matching logic, and added the unit tests in `types_test.go`. 

Ready for review!

**Comment by nako-ruru** [comment]
### 💡 A Side Note Regarding RFC 4918 Compliance & `Content-Type`

While debugging this issue, I noticed another potential edge case related to the `useStandardProps` implementation that I wanted to share as an FYI (though I chose **not** to include it in this PR to minimize risk).

When `useStandardProps` is enabled, rclone sends an XML payload via `standardProps`. According to **RFC 4918 (Section 8.1)** and standard **HTTP/1.1 (RFC 7231)**, any HTTP request carrying a message body *should* explicitly declare a `Content-Type` header (e.g., `text/xml` or `application/xml`). 

Currently, rclone sends this XML payload without setting the `Content-Type` header. 
- My specific WebDAV server library strictly requires this and immediately rejects the request with an **HTTP 400 Bad Request**. 
- Out of curiosity, I asked Gemini about this behavior. It pointed out that theoretically, some strict RFC-compliant servers might even throw an **HTTP 415 Unsupported Media Type** because the header is missing or non-XML. 

Either way, it's a bit of a "compatibility lottery" depending on the backend implementation.

**My Workaround:**
To avoid breaking other existing setups or risking regressions within rclone's request-sending pipeline, I solved this safely by adding a custom middleware on my server to sniff the `PROPFIND` payload and automatically inject the missing `Content-Type` header.

I'm leaving rclone's request headers untouched in this PR to keep it hyper-focused on the response status parsing, but I wanted to bring this up to the maintainers in case you want to enforce standard `Content-Type: text/xml` headers for XML-bodied requests in a future refactor.

**Comment by ncw** [comment]
@nako-ruru We don't explicitly set the Content-Type for any of the xml calls, but CallXML does do that if we don't pass a body in

https://github.com/rclone/rclone/blob/ffd36fc2d7bd8ac15d1e554fff632991fbdb8e82/lib/rest/rest.go#L538-L540

https://github.com/rclone/rclone/blob/ffd36fc2d7bd8ac15d1e554fff632991fbdb8e82/lib/rest/rest.go#L553-L558

So all the places we set `opts.Body` or `Body:` need checking to see if we need to add a `ContentType` to the opts.

I'd be happy to receive that as a follow up PR.

I think this PR is ready to merge - I just need to wait for the tests to run. The `govulncheck` warning I will fix in a bit.

**Review by ncw — APPROVED** [review]
This looks great now - thank you :-)
